harvard-lts/jstor_aspace_publisher
# SS-159: Implement publishing tasks in publisher

**Implement publishing tasks in publisher**
* * *

**JIRA Ticket**: https://jira.huit.harvard.edu/browse/SS-159

# What does this Pull Request do?
adds s3 publishing capability to the publisher

# How should this be tested?

1) run the integration test locally `curl -k "https://localhost:25003/integration"` or on dev `https://ltsds-cloud-dev-1.lib.harvard.edu:25003/integration`

2) go into the container and check the logs. you should see something like this:

`{"event": "**************JStor Publisher: Do Task**************", "logger": "app", "level": "error", "timestamp": "2023-
[..]
{"event": "running integration test", "logger": "app", "level": "info", "timestamp": "2023-02-06T10:04:05.653956Z"}
{"event": "configfile: /home/jstorforumadm/harvestjobs_test.json", "logger": "app", "level": "info", "timestamp": "2023-02-06T10:04:05.655241Z"}
{"event": "publishing to S3", "logger": "app", "level": "info", "timestamp": "2023-02-06T10:04:05.661533Z"}
{"event": "looking in current path: /tmp/harvested/warren_test", "logger": "app", "level": "info", "timestamp": "2023-02-06T10:04:05.662945Z"}
{"event": "Publishing set: warren_test", "logger": "app", "level": "info", "timestamp": "2023-02-06T10:04:05.666396Z"}
{"event": "Uploading: /tmp/harvested/warren_test/8000245145_test.xml to warren_test/8000245145_test.xml", "logger": "app", "level": "info", "timestamp": "2023-02-06T10:04:05.668332Z"}
{"event": "looking in current path: /tmp/transformed/warren_test", "logger": "app", "level": "info", "timestamp": "2023-02-06T10:04:05.954654Z"}

3) confirm that the record is in s3 using the aws cli, ex: `aws s3 ls s3://harvard.via.dev/  --profile=via-dev `
you should see a directory with the "warren_test" repo with the timestamp of when you uploaded it and it should have several xml files.



# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? no
- integration tests? y

# Interested parties
Tag (@ mention) interested parties: @awoods
